module/lua/llimits.h

```diff
@@ -126,7 +126,11 @@ typedef LUAI_UACNUMBER l_uacNumber;
  * Minimum amount of available stack space (in bytes) to make a C call.  With
  * gsub() recursion, the stack space between each luaD_call() is 1256 bytes.
  */
+#if defined(ZFS_DEBUG) && defined(__FreeBSD__) && defined(_KERNEL)
+#define LUAI_MINCSTACK		4224 /* need some extra bytes in debug build */
+#else
 #define LUAI_MINCSTACK		4096
+#endif
```

Comment:
Making the debug build behave slightly differently in terms of maximum depth doesn't seem desirable to me.  We're only talking about 128 bytes so perhaps you could instead increase it to `4224` for all FreeBSD builds to be consistent?  With a comment explaining why of course.  That said, I'm OK with this change as is if in practice it's a non issue on FreeBSD.

Reply:
I wonder if the same issue occurs on Linux when the kernel stack size is increased similarly. I will see if I can find a way to test that.

Reply:
Seems that the stack size is a hard coded constant in Linux, so I'm not going to start messing with that.

Reply:
There are a lot of ways to build the Linux kernel, including increasing the stack size, but practically speaking we have to do our testing with the configurations used by stock distribution kernels.  And for those kernels 4096 is sufficient even with ZFS debugging enabled.

Reply:
Was there a test bot with a KASAN kernel? I don't think that makes the stacks big enough to hit this error case still, just came across this while researching and it rang a bell:
https://github.com/torvalds/linux/blob/master/arch/x86/include/asm/page_64_types.h#L9-L16

Reply:
Thanks for pointing that out.  Unfortunately, we don't have a bot for that yet.